examples/weekly-summary-email/steps/run/cmd/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/inngest/inngestgo"
+	"github.com/inngest/inngestgo/actionsdk"
+	sendgrid "github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+)
+
+const (
+	FromEmail = "noreply@example.com"
+	FromName  = "Your Service"
+	Subject   = "Your [service] weekly summary"
+	// TemplateID is the sendgrid template that we're sending
+	TemplateID = "d-c6dcf1f72bdd4beeb15a9aa6c72fcd2c"
+)
+
+func main() {
+	n, err := Run(context.Background())
+	if err != nil {
+		actionsdk.WriteResult(&actionsdk.Result{
+			Body: map[string]any{
+				"sent":  n,
+				"error": err.Error(),
+			},
+			Status: 500, // A status of 5xx retries the function.
+		})
+		os.Exit(1)
+	}
+
+	actionsdk.WriteResult(&actionsdk.Result{
+		Body: map[string]any{
+			"sent": n,
+		},
+		Status: 200,
+	})
+}
+
+// Run sends a weekly summary email for each account fetched from the
+// SummaryFetcher.
+//
+// You can access internal APIs, databases, and other data sources as
+// you'd expect from normal functions to fetch data in the real world.
+//
+// This returns the number of summaries sent.
+func Run(ctx context.Context) (int, error) {
+	all, err := NewSummaryFetcher().Fetch(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	for n, s := range all {
+		// Send this summary.
+		//
+		// An alternative strategy here is to send another event
+		// that triggers a new function to send this summary.  This
+		// ensures that retries are local to the specific summary
+		// email being sent.
+		if err := Send(ctx, s); err != nil {
+			return n, err
+		}
+
+		// Send a "summary/user.fetched" event to Inngest.  We can
+		// then create a new function that is triggered by this event
+		// to send the email vs doing it here.
+		//
+		// This gives logs per-user, allows retries per-user, and allows
+		// us to handle SendAt natively via waits within our function:
+		// https://www.inngest.com/docs/functions/step-functions#after-configuration
+		inngestgo.NewClient(os.Getenv("INNGEST_EVENT_KEY")).Send(ctx, inngestgo.Event{
```

Comment:
Is this a secret we need to be passing in to the function? Should it be in `.env`?